src/libstore/local-store.cc

```diff
@@ -446,10 +446,20 @@ LocalStore::LocalStore(const Params & params)
 }
 
 
-LocalStore::LocalStore(std::string scheme, std::string path, const Params & params)
-    : LocalStore(params)
+LocalStore::LocalStore(
+    std::string_view scheme,
+    std::optional<PathView> path,
+    const Params & _params)
+    : LocalStore([&]{
+        // ?root=... overrides path
```

Comment:
```suggestion
        // Default `?root` from `path` if non set
```

(the current comment wans't really clear to me)